learningresources/api.py

```diff
@@ -222,30 +225,71 @@ def get_resource(resource_id, user_id):
     return resource
 
 
-def create_static_asset(course_id, file_handle):
+def create_static_asset(course_id, handle):
     """
-    Create a static asset for a given course.
+    Create a static asset.
+    Args:
+        course_id (int): learningresources.models.Course pk
+        handle (django.core.files.File): file handle
+    Returns:
+        learningresources.models.StaticAsset
+    """
+    with transaction.atomic():
+        return StaticAsset.objects.create(course_id=course_id, asset=handle)
 
-    Warning:
-        This takes and open file handle and does not close it. You
-        will need to handle the opening and closing the
-        ``file_handle`` outside of this method.
 
-    Raises:
-        ValueError: If a closed handle is passed in
+def _subs_filename(subs_id, lang='en'):
+    """
+    Generate proper filename for storage.
+
+    Function copied from:
+    edx-platform/common/lib/xmodule/xmodule/video_module/transcripts_utils.py
 
     Args:
-        course_id (int): Primary key of the Course to add the asset to.
-        file_handle (django.core.files.File):
-            An open file handle to save to the model.
+        subs_id (str): Subs id string
+        lang (str): Locale language (optional) default: en
 
     Returns:
-        learningresources.models.StaticAsset
+        filename (str): Filename of subs file
     """
-    course = Course.objects.get(id=course_id)
-    with transaction.atomic():
-        static_asset, _ = StaticAsset.objects.get_or_create(
-            course=course,
-            asset=file_handle
-        )
-    return static_asset
+    if lang in ('en', "", None):
+        return u'subs_{0}.srt.sjson'.format(subs_id)
+    else:
+        return u'{0}_subs_{1}.srt.sjson'.format(lang, subs_id)
+
+
+def get_video_sub(xml):
+    """
+    Get subtitle IDs from <video> XML.
+
+    Args:
+        xml (lxml.etree): xml for a LearningResource
+    Returns:
+        sub string: subtitle string
+    """
+    subs = xml.xpath("@sub")
+    # It's not possible to have more than one.
+    if len(subs) == 0:
+        return ""
+    return _subs_filename(subs[0])
+
+
+def import_static_assets(course, path):
```

Comment:
I think it was because it's directly creating model instances within the `learningresources` app, so it seemed to fit better there.